src/EVM/SMT.hs

```diff
@@ -107,6 +112,26 @@ declareIntermediates bufs stores =
     encodeStore n expr =
        fromLazyText ("(define-const store" <> (T.pack . show $ n) <> " Storage ") <> exprToSMT expr <> ")"
 
+isAbstractStorage :: Expr Storage -> Bool
+isAbstractStorage = getAny . foldExpr go (Any False)
+  where
+    go :: Expr a -> Any
```

Comment:
This `Any` is pretty cute, didn't know about it before :)

Reply:
Yes! I found out while I was trying to write this function, and was looking for a monoid for Bool. There's an `All` too :)